mmdet/models/dense_heads/anchor_head.py

```diff
@@ -633,7 +639,7 @@ def _get_bboxes_single(self,
                 scores = cls_score.softmax(-1)
             bbox_pred = bbox_pred.permute(1, 2, 0).reshape(-1, 4)
             nms_pre = cfg.get('nms_pre', -1)
-            if nms_pre > 0 and scores.shape[0] > nms_pre:
+            if nms_pre > 0:
```

Comment:
Setting K of TopK to dynamic could be aligned with this [PR #4684](https://github.com/open-mmlab/mmdetection/pull/4684/files) for the following reason:

1. Only create one `nms_pre` constant tensor;
2. No change to original logic for training and testing in PyTorch.